packages/esm-patient-orders-app/src/order-cancellation-form/cancel-order-form.component.tsx

```diff
@@ -0,0 +1,182 @@
+          title: t(`errorCancellingOrder', 'Error cancelling order`),
+          kind: 'error',
+          subtitle: err?.message,
+        });
+      },
+    );
+  }, []);
+
+  return (
+    <Form className={styles.form}>
+      <div className={styles.grid}>
+        <Stack>
+          <section>
+            <h4 style={{ marginBottom: '1rem' }}>{order.display}</h4>
+          </section>
+          <section>
+            <Controller
+              name="cancellationDate"
+              control={control}
+              render={({ field: { onChange, value } }) => (
+                <div className={styles.row}>
+                  <DatePicker
+                    id="vaccinationExpiration"
+                    className="vaccinationExpiration"
+                    minDate={dayjs().startOf('day').format('DD/MM/YYYY')}
+                    dateFormat="d/m/Y"
+                    datePickerType="single"
+                    value={value}
+                    onChange={([date]) => onChange(date)}
+                    autocomplete="off"
+                  >
+                    <DatePickerInput
+                      id="date-picker-calendar-id"
+                      placeholder="dd/mm/yyyy"
+                      labelText={t('cancellationDate', 'Cancellation date')}
+                      type="text"
+                    />
+                  </DatePicker>
+                </div>
+              )}
+            />
+          </section>
+          <section>
+            <Controller
+              name="reasonForCancellation"
+              control={control}
+              render={({ field: { onChange, value } }) => (
+                <div className={styles.row}>
+                  <TextArea
+                    type="text"
+                    id="reasonForCancellation"
+                    labelText={t('reasonForCancellation', 'Reason for cancellation')}
+                    value={value}
+                    onChange={(evt) => onChange(evt.target.value)}
+                  />
+                </div>
+              )}
+            />
+          </section>
+        </Stack>
+      </div>
+
+      {showErrorNotification && (
+        <Column className={styles.errorContainer}>
+          <InlineNotification
+            lowContrast
+            title={t('error', 'Error')}
+            subtitle={t('pleaseFillField', 'Please fill at least one field') + '.'}
+            onClose={() => setShowErrorNotification(false)}
+          />
+        </Column>
+      )}
+
+      <ButtonSet className={isTablet ? styles.tablet : styles.desktop}>
+        <Button className={styles.button} kind="secondary" onClick={() => closeWorkspace()}>
+          {t('discard', 'Discard')}
+        </Button>
+        <Button
+          className={styles.button}
+          kind="primary"
+          onClick={handleSubmit(cancelOrderRequest, onError)}
+          disabled={isSubmitting}
+          type="submit"
+        >
+          {t('saveAndClose', 'Save and close')}
```

Comment:
Because we already have the `isSubmitting` state, could be nice to tack something like this on here:

```tsx
{isSubmitting ? (
  <InlineLoading description={t('saving', 'Saving') + '...'} />
) : (
  {t('saveAndClose', 'Save and close')}
)}
```

Doing so would require the following style overrides, however:

```scss
.button {
  :global(.cds--inline-loading) {
    min-height: 1rem !important;
  }

  :global(.cds--inline-loading__text) {
    font-size: unset !important;
  }
}
```